url_filter/backends/django.py

```diff
@@ -113,7 +113,8 @@ def filter_by_specs(self, queryset):
             queryset = queryset.exclude(**{lookup: value})
 
         to_many = self._is_any_to_many()
-        return queryset.distinct() if to_many and (include or exclude) else queryset
+        is_distinct = queryset.distinct()
```

Comment:
as per docs `.distinct()` [returns another queryset](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#django.db.models.query.QuerySet.distinct) and using `not is_distinct` should be equivalent to `not is_distinct.__bool__()` which internally [fetches all results](https://github.com/django/django/blob/7e3bf2662b5dedeb47856adc18a9378e2d10a599/django/db/models/query.py#L276-L278). therefore this will have another db hit. am I missing something?

from quick code glance this might achieve what you are looking for although dont think this is public query api which can make it fragile.

```suggestion
        is_distinct = bool(queryset.query.distinct_fields)
```